lib/hooks/useAffiliate.ts

```diff
@@ -0,0 +1,26 @@
+import { useRouter } from 'next/router'
+import { useState, useEffect } from 'react'
+
+export const LOCAL_STORAGE_AFFILIATE_KEY = 'rcvr_affiliate'
+
+export const useAffiliate = () => {
```

Comment:
After thinking about this again it might be problematic to store this in a localStorage.

Scenario:

- I visit recover with an affiliate link of cologne
- A couple of days later i visit recover in another city with a link that is not an affiliate
- Result: I see the affiliate of cologne. Which might be bad if you see it in Düsseldorf 😉.

Reply:
Right! Should I take `useOwner` and find out the affiliate case with it?

Reply:
If I send someone a link to register to recover as an owner, that person should see the affiliate that is in that link. In that case, we don't have an owner yet 😢 . And then `useOwner` will not work.

So the only solution that I can think of right now has 2 steps:

- show an affiliate if it is in the owner if `useOwner` returns an owner with affiliate
- if the link has query params, we need to keep those params when we switch the language.

Reply:
Hm, damm 😢. I had to type a `Link` component to make the language switch happen properly. I can add the query params there I think so it can be persisted while navigating. But I'll also check if that is a feature nextjs supports.

Reply:
Maybe you have another Idea or my view on this is wrong?

Reply:
@bitboxer nah you are right because `localStorage` is the same for the same user + browser unless is cleaned. Even in phones (specially) you are right 😆. So I'll look into the query string being passed with links. I think is possible because of what I was saying before with the Link component. Will have to look though to see if we add query strings at all (haven't noticed) but I should check that too to not break it.

Reply:
Actually, we can use [`sessionStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage) after all since that's our use case. Has [good support](https://caniuse.com/?search=sessionstorage) too.

Will try to get the data from the owner when there and if not from the `sessionStorage`. Persisting the query string might work but is fragile I think.

Reply:
I'll resolve this. See `lib/useAffiliate.ts` and add comments there if you find we still need to do something.